activator/activator.go

```diff
@@ -0,0 +1,90 @@
+package activator
+
+import (
+	"os"
+
+	guuid "github.com/google/uuid"
+
+	"v9_deployment_manager/database"
+	"v9_deployment_manager/log"
+	"v9_deployment_manager/worker"
+)
+
+type Activator struct {
+	driver *database.Driver
+}
+
+func CreateActivator(driver *database.Driver) (*Activator, error) {
+	return &Activator{
+		driver: driver,
+	}, nil
+}
+
+func (a *Activator) Activate(compID *worker.ComponentID, worker *worker.V9Worker) error {
+	// Get random tar name
+	tarName := guuid.New().String()
+	fullRepoName := compID.User + "/" + compID.Repo
+	// Get Repo Contents
+	log.Info.Println("Cloning Repo...")
+	clonedPath, err := cloneRepo(fullRepoName)
+	if err != nil {
+		log.Error.Println("Error cloning repo:", err)
+		return err
+	}
+	defer os.RemoveAll(clonedPath) // clean up
+
+	//FIXME git checkout HEAD/hash
```

Comment:
How hard is fixing this?

Reply:
Unsure, I'm using go-git to perform the clone. It's not immediately clear to me if the clone implementation defaults to HEAD. I'll look into this.

Reply:
Do we want to be able to clone from a specific hash? If so that could make this more difficult.